tests/ui/resolve/issue-109343.rs

```diff
@@ -0,0 +1,13 @@
+#![crate_type = "lib"]
```

Comment:
maybe it is enough to keep only one test case? 🤔

Reply:
Yes, let's keep only the minimized example from https://github.com/rust-lang/rust/pull/109602#issuecomment-1492161447, others looks equivalent, just with more distractions.